src/library/Uno.Material/Controls/Chips/ChipGroup.cs

```diff
@@ -0,0 +1,332 @@
+
+				container.IsCheckedChanged += OnItemIsCheckedChanged;
+				container.Click += OnItemClick;
+				container.Checked += OnItemChecked;
+				container.Unchecked += OnItemUnchecked;
+			}
+		}
+
+		protected override void ClearContainerForItemOverride(DependencyObject element, object item)
+		{
+			base.ClearContainerForItemOverride(element, item);
+
+			if (element is Chip container)
+			{
+				container.ClearValue(Chip.IsCheckedProperty);
+
+				container.Thumbnail = null;
+				container.ThumbnailTemplate = null;
+				container.CanRemove = false;
+
+				container.IsCheckedChanged -= OnItemIsCheckedChanged;
+				container.Click -= OnItemClick;
+				container.Checked -= OnItemChecked;
+				container.Unchecked -= OnItemUnchecked;
+			}
+		}
+
+		private void OnItemClick(object sender, RoutedEventArgs e) => RaiseItemEvent(ItemClick, sender);
+
+		private void OnItemChecked(object sender, RoutedEventArgs e) => RaiseItemEvent(ItemChecked, sender);
+
+		private void OnItemUnchecked(object sender, RoutedEventArgs e) => RaiseItemEvent(ItemUnchecked, sender);
+
+		private void RaiseItemEvent(ChipItemEventHandler handler, object originalSender)
+		{
+			if (originalSender is Chip container)
+			{
+				handler?.Invoke(this, new ItemClickEventArgs(ItemFromContainer(container)));
+			}
+		}
+
+		private void EnforceSelectionMode()
+		{
+			if (ItemsPanelRoot == null) return;
+
+			if (SelectionMode == ChipSelectionMode.None)
+			{
+				UpdateItemsIsCheckable();
+				UpdateSelection(default);
+			}
+			else if (SelectionMode == ChipSelectionMode.Single)
+			{
+				UpdateItemsIsCheckable();
+
+				// either one is selected or none are selected
+				var selectedContainers = GetItemContainers().Where(x => x.IsChecked == true).ToArray();
+				if (selectedContainers.Length > 1)
+				{
+					foreach (var container in selectedContainers)
+					{
+						container.SetIsCheckedSilently(false);
+					}
+					UpdateSelection(default);
+				}
+				else
+				{
+					UpdateSelection(selectedContainers);
+				}
+			}
+			else if (SelectionMode == ChipSelectionMode.Multiple)
+			{
+				UpdateItemsIsCheckable();
+				UpdateSelection(default);
+			}
+
+			void UpdateItemsIsCheckable()
+			{
+				foreach (var container in GetItemContainers())
+				{
+					container.IsCheckable = SelectionMode != ChipSelectionMode.None;
+				}
+			}
+		}
+
+		private void UpdateSelection(Chip[] newlySelectedContainers, bool forceClearOthersSelection = false)
+		{
+			if (ItemsPanelRoot == null) return;
+			if (_isSynchronizingSelection) return;
+
+			try
+			{
+				_isSynchronizingSelection = true;
+
+				var selectedItems = new List<object>();
+				foreach (var container in GetItemContainers())
+				{
+					if (container.IsChecked == true)
+					{
+						if (ShouldClearSelection())
+						{
+							container.SetIsCheckedSilently(false);
+						}
+						else
+						{
+							selectedItems.Add(ItemFromContainer(container));
+						}
+
+						bool ShouldClearSelection()
```

Comment:
How does this work? why not an if statement?

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions